### PR TITLE
Remove circuit_duration_check from RunRequest, add max_circuit_duration_over_t2

### PR DESCRIPTION
**Pull request by ollityrkko**
Adapt to CoCoS v23

### CODE REVIEW

File: src/iqm/iqm_client/iqm_client.py [hunk 1]
```diff
@@ -804,8 +805,8 @@ def submit_circuits(
                 Note: This field should always be ``None`` in normal use.
             calibration_set_id: ID of the calibration set to use, or ``None`` to use the latest one
             shots: number of times ``circuits`` are executed, value must be greater than zero
-            circuit_duration_check: whether to enable max circuit duration criteria for disqualification
-            heralding_mode: Heralding mode to use during the execution.
```
Comment (jkotilahti) [inline src/iqm/iqm_client/iqm_client.py:808]: It looks like the `heralding_mode` docstring was accidentally removed. It should be restored.

File: src/iqm/iqm_client/iqm_client.py [hunk 2]
```diff
@@ -804,8 +805,8 @@ def submit_circuits(
...
+            max_circuit_duration_over_t2: Circuits are disqualified on the server if they are longer than this ratio
+                of the T2 time of the qubits.
```
Comment (jkotilahti) [inline src/iqm/iqm_client/iqm_client.py:809]: The special meaning of `max_circuit_duration_over_t2=0.0` (and maybe also exact meaning of `max_circuit_duration_over_t2=None`) could be documented here as well.

**Review by jkotilahti — COMMENTED** [review]
Other than the minor issues with docstrings, it looks good to me. I will approve after also the Cocos changes can be checked.

File: src/iqm/iqm_client/iqm_client.py (hunk 1 above)
Comment (ollityrkko) [reply src/iqm/iqm_client/iqm_client.py:808]: Oops. Fixed.

File: src/iqm/iqm_client/iqm_client.py (hunk 2 above)
Comment (ollityrkko) [reply src/iqm/iqm_client/iqm_client.py]: Added more documentation here.

File: src/iqm/iqm_client/iqm_client.py [hunk 3]
```diff
@@ -804,7 +805,9 @@ def submit_circuits(
                 Note: This field should always be ``None`` in normal use.
             calibration_set_id: ID of the calibration set to use, or ``None`` to use the latest one
             shots: number of times ``circuits`` are executed, value must be greater than zero
-            circuit_duration_check: whether to enable max circuit duration criteria for disqualification
+            max_circuit_duration_over_t2: Circuits are disqualified on the server if they are longer than this ratio
+                of the T2 time of the qubits. Setting this value to ``0.0`` turns off circuit duration checking.
+                The defaul value ``None`` instructs server to use server's default value in the checking.
```
Comment (jkotilahti) [inline src/iqm/iqm_client/iqm_client.py:810]:
```suggestion
                The default value ``None`` instructs server to use server's default value in the checking.
```

**Review by jkotilahti — APPROVED** [review]

**Pull request merged**